src/port1.0/portfetch.tcl

```diff
@@ -607,11 +607,20 @@ proc portfetch::fetch_addfilestomap {filemapname} {
     }
 }
 
+# Utility function to error out if Xcode is needed
+proc portfetch::checkxcode {} {
+    global use_xcode developer_dir
+    if {$use_xcode eq "yes" && $developer_dir eq ""} {
+        return -code error "This port requires Xcode, which was not found on your system."
+    }
+}
+
 # Initialize fetch target and call checkfiles.
 proc portfetch::fetch_init {args} {
     variable fetch_urls
 
     portfetch::checkfiles fetch_urls
+    portfetch::checkxcode
```

Comment:
@neverpanic By "pre-fetch" do you mean like this? I don't fully understand the portindex problem yet.

Reply:
Yes, that'll work.

As for the portindex problem, try the following:
1. Create a local Portfile repository as described in https://guide.macports.org/#development.local-repositories
2. Write a port that contains `error "testing"` outside of a phase block
3. Run `portindex` as described in the guide. It will fail to index the port, and the generated Portindex will not contain any information about that port.

Now, change that port such that the `error "testing"` statement is in a `pre-fetch { ... }` block. If you run portindex again, it should successfully parse the port and add an entry to the Portindex. You still won't be able to install the port (because it's fetch phase will fail), but you will be able to run `port info` on it.